.github/workflows/e2e_hardware.yml

```diff
@@ -0,0 +1,29 @@
+name: e2e_hardware
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+jobs:
+  e2e-test:
+    if: contains(github.event.pull_request.labels.*.name, 'hardware')
+    name: E2E test on hardware
+    runs-on: self-hosted
```

Comment:
I think we have to be more specific here. 
```
runs-on: [self-hosted, linux, x64, hugepages]
```

do we need anything else ? I think we just need hugepages right?

Reply:
Yes, only hugepages needs to be run with hardware and we can add more labels to this **runs-on** action to precisely select runner instead of running test on any. We only need to use the same labels within runner settings.

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on